Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.h

```diff
@@ -86,6 +87,8 @@ class ECMAScriptFunctionObject final : public FunctionObject {
 
     FunctionKind kind() const { return m_kind; }
 
+    void set_script_or_module(ScriptOrModule script_or_module) { m_script_or_module = move(script_or_module); }
```

Comment:
https://github.com/SerenityOS/serenity/pull/10478#discussion_r731790872

> This appears to be unused, and I don't think replacing a function's [[ScriptOrModule]] is something the spec will ever do.

Reply:
This is used on step 3.11 of this algorithm: https://html.spec.whatwg.org/multipage/webappapis.html#getting-the-current-value-of-the-event-handler

> 11. Set function.[[ScriptOrModule]] to null.

The spec gives this reason for doing so:

> This is done because the default behavior, of associating the created function with the nearest [script](https://html.spec.whatwg.org/multipage/webappapis.html#concept-script) on the stack, can lead to path-dependent results. For example, an event handler which is first invoked by user interaction would end up with null [[ScriptOrModule]] (since then this algorithm would be first invoked when the [active script](https://html.spec.whatwg.org/multipage/webappapis.html#active-script) is null), whereas one that is first invoked by dispatching an event from script would have its [[ScriptOrModule]] set to that script.

> Instead, we just always set [[ScriptOrModule]] to null. This is more intuitive anyway; the idea that the first script which dispatches an event is somehow responsible for the event handler code is dubious.

> In practice, this only affects the resolution of relative URLs via [import()](https://tc39.es/ecma262/#sec-import-calls), which consult the [base URL](https://html.spec.whatwg.org/multipage/webappapis.html#concept-script-base-url) of the associated script. Nulling out [[ScriptOrModule]] means that [HostResolveImportedModule](https://html.spec.whatwg.org/multipage/webappapis.html#hostresolveimportedmodule(referencingscriptormodule,-modulerequest)) and [HostImportModuleDynamically](https://html.spec.whatwg.org/multipage/webappapis.html#hostimportmoduledynamically(referencingscriptormodule,-modulerequest,-promisecapability)) will fall back to the current settings object's [API base URL](https://html.spec.whatwg.org/multipage/webappapis.html#api-base-url).

Reply:
Cool, let's add that URL with a small comment then :)